R/plumber.R

```diff
@@ -853,10 +850,23 @@ Plumber <- R6Class(
     set404Handler = function(fun){
       private$notFoundHandler <- fun
     },
+    #' @description Sets the handler that gets called if an
+    #' incoming request uses an unallowed method.
+    #'
+    #' See also: [pr_set_405()]
+    #' @param fun a handler function.
+    #' @examples
+    #' \dontrun{
+    #' pr <- pr()
+    #' pr$set405Handler(function(req, res) {cat(req$PATH_INFO)})
+    #' }
+    set405Handler = function(fun){
+      private$unallowedMethodHandler <- fun
```

Comment:
`set404Handler` should maybe do the same (to avoid errors that aren't caught until runtime)?

```suggestion
      stopifnot(is.function(fun))
      args <- names(formals(fun))
      if (!all(c("req", res") %in% args)) {
        stop("fun must have formal arguments for req and res")
      }
      private$unallowedMethodHandler <- fun
```

Reply:
There a couple other plumber functions that take a function as input and are executed at runtime with potential failure. This might be material for another PR

Reply:
@meztez Would you like / have time to do a PR for the other method parameter checks?

Reply:
Eventually, I still have the regular job thing to do 😄